### PR TITLE
Provide authorization

### PR DESCRIPTION
**Pull request by esimonov**

### CODE REVIEW

File: authorization/policy.csv [hunk 1]
```diff
@@ -0,0 +1,4 @@
+p, anonymous, 59fd884d8f6b180001f5b4e2:/auth/login, POST
```
Comment (erudenko) [inline authorization/policy.csv]: What is it for?

File: web/api/router.go [hunk 2]
```diff
@@ -67,6 +69,18 @@ func WebRouterPrefixOption(prefix string) func(*Router) error {
 	}
 }
 
+// AuthorizationOption is an option to set casbin enforcer for authorization.
+func AuthorizationOption(params []interface{}) func(r *Router) error {
```
Comment (erudenko) [inline web/api/router.go]: do we need it?

File: web/api/sms_login.go [hunk 3]
```diff
@@ -112,6 +112,17 @@ func (ar *Router) PhoneLogin() http.HandlerFunc {
 			ar.Error(w, ErrorAPIAppAccessTokenNotCreated, http.StatusInternalServerError, err.Error(), "LoginWithPassword.loginUser")
 			return
 		}
+
+		azi := authzInfo{
```
Comment (erudenko) [inline web/api/sms_login.go]: why it's here?

File: web/api/authorization.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	appID       string
+	tokenStr    string
+	resourceURI string
+	method      string
+}
+
+// Authorize checks if user has an access to the requested resource.
+// If error happens, writes it to ResponseWriter.
+// Also, writes an error on failed authorization.
+func (ar *Router) Authorize(w http.ResponseWriter, azi authzInfo) error {
```
Comment (erudenko) [inline web/api/authorization.go]: we don't have user->resource auth scheme

File: authorization/policy.csv (hunk 1 above)
Comment (esimonov) [reply authorization/policy.csv]: It is for allowing anonymous users to login into the app with id 59fd884d8f6b180001f5b4e2.

File: web/api/router.go (hunk 2 above)
Comment (esimonov) [reply web/api/router.go]: Because do not want to make authorization mandatory, don't we?

File: web/api/authorization.go (hunk 4 above)
Comment (esimonov) [reply web/api/authorization.go]: It's a role->resource scheme essentially.

File: web/api/authorization.go [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+	if accessGranted := authorizer.Enforce(sub, obj, act); !accessGranted {
+		err := fmt.Errorf("Access denied")
+		ar.Error(w, ErrorAPIAppAccessDenied, http.StatusForbidden, err.Error(), "Authorizer.AccessDenied")
+		return err
+	}
+	return nil
+}
+
+func (ar *Router) initInternalAuthorizer(app model.AppData) (*casbin.Enforcer, error) {
```
Comment (erudenko) [inline web/api/authorization.go]: Brilliant

**Pull request merged**